src/algebra/roption.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2019 Joe Cool. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Hoang Le Truong.
+
+If α is a comm_monoid  then roption α is a comm_monoid. 
+
+-/
+
+import   algebra.module data.pfun
+
+
+universes u v 
+variables {α : Type u} {β : Type v} 
+local attribute [instance] classical.prop_decidable
+
+
+namespace roption
+noncomputable theory
+
+instance [has_add α]: has_add (roption α) := ⟨ λ x y, ⟨x.dom ∧ y.dom, λ h, x.get (h.1)+ y.get (h.2)  ⟩  ⟩
+def add_def [has_add α] (x y: roption α) : x+y =⟨x.dom ∧ y.dom, λ h, x.get (h.1)+ y.get (h.2)  ⟩ :=rfl
+
+instance  [has_zero α ] :  has_zero (roption (α ))  :=⟨some (0:α)⟩
+def zero_def [has_zero α] : (0:roption α ) =⟨true, λ _, (0:α) ⟩ :=rfl
+
+instance   [has_one α ]:  has_one (roption (α ))  :=⟨some (1:α)⟩
```

Comment:
Formatting: remove duplicate spaces after `instance` and before `:=`, insert space before `:`, and after `:=`.

Reply:
More generally, PRs will have formatting corrected before they are accepted, and while everyone will be pretty patient about reminding authors about mistakes (I'm terrible), it saves everyone time if you invest effort in cleaning up formatting first.

Reply:
I changed it.